VocabularyTrainer/AddWord/AddNewWordViewController.swift

```diff
@@ -0,0 +1,230 @@
+    
+    private func setUpConstraints() {
+        backButton.topAnchor.constraint(equalTo: view.topAnchor, constant: 16).isActive = true
+        backButton.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 16).isActive = true
+        
+        addNewWordLabel.centerXAnchor.constraint(equalTo: view.centerXAnchor).isActive = true
+        addNewWordLabel.topAnchor.constraint(equalTo: backButton.bottomAnchor, constant: 16).isActive = true
+        
+        newWordTextField.topAnchor.constraint(equalTo: addNewWordLabel.bottomAnchor, constant: 16).isActive = true
+        newWordTextField.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 24).isActive = true
+        newWordTextField.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -24).isActive = true
+        
+        translationTextField.topAnchor.constraint(equalTo: newWordTextField.bottomAnchor, constant: 16).isActive = true
+        translationTextField.leadingAnchor.constraint(equalTo: newWordTextField.leadingAnchor).isActive = true
+        translationTextField.trailingAnchor.constraint(equalTo: newWordTextField.trailingAnchor).isActive = true
+        
+        addButton.topAnchor.constraint(equalTo: translationTextField.bottomAnchor, constant: 24).isActive = true
+        addButton.leadingAnchor.constraint(equalTo: translationTextField.leadingAnchor).isActive = true
+        addButton.trailingAnchor.constraint(equalTo: translationTextField.trailingAnchor).isActive = true
+    }
+    
+    private func setUpInterfaceStyleUI() {
+        switch traitCollection.userInterfaceStyle {
+        case .light, .unspecified:
+            addNewWordLabel.textColor = .black
+            
+            newWordTextField.backgroundColor = .white
+            newWordTextField.textColor = .black
+            
+            translationTextField.backgroundColor = .white
+            translationTextField.textColor = .black
+        case .dark:
+            addNewWordLabel.textColor = .white
+            
+            newWordTextField.backgroundColor = .black
+            newWordTextField.textColor = .white
+            
+            translationTextField.backgroundColor = .black
+            translationTextField.textColor = .white
+        }
+    }
+    
+    private func addButtonIsEnabled(_ isEnabled: Bool) {
+        if isEnabled {
+            UIView.animate(withDuration: 0.3) { [weak self] in
+                self?.addButton.backgroundColor = BackgroundColor.mediumSpringBud
+            }
+            addButton.isEnabled = isEnabled
+        } else {
+            UIView.animate(withDuration: 0.3) { [weak self] in
+                self?.addButton.backgroundColor = UIColor(red: 207/255, green: 207/255, blue: 205/255, alpha: 0.4)
+            }
+            addButton.isEnabled = isEnabled
+        }
+    }
+    
+    private func wordAddedNotification() {
+        NotificationCenter.default.post(name: NSNotification.Name(rawValue: "wordAdded"), object: nil)
```

Comment:
Thanks for adding this!

Although I am not a big fan of `NotificationCenter` as these are always "too global" for my taste, I think it's a good use case here.

I'm thinking about separating the data layer in general a little bit, then we might refactor this as well somehow. Maybe having combine subscribing to properties in the data layer.

One thing: Could you add an extension to `NSNotification.Name` so that we do not have to use the hardcoded strings inside here and the observing part in `LanguageScreenViewController`?

Reply:
Yeah. I'll do that :)

Reply:
Thanks! After that I'll approve and merge, looks great, thank you!

Reply:
Added :)